### DISCUSSION TITLE
'tokio-runtime-worker' has overflowed its stack

### DISCUSSION THREAD

**Discussion by 25sumitkr** [tokio-rs/tokio#3062] [Q&A] [answered]
**Version**
│   │   │   │   ├── tokio v0.2.22
│   │   │   │   │   └── tokio-macros v0.2.5
│   │   │   │   ├── tokio-util v0.3.1
│   │   │   │   │   └── tokio v0.2.22 (*)
│   │   │   ├── tokio v0.2.22 (*)
│   │   │   ├── tokio v0.2.22 (*)
│   │   │   ├── tokio-rustls v0.14.1
│   │   │   │   ├── tokio v0.2.22 (*)
│   │   ├── tokio v0.2.22 (*)
│   │   ├── tokio-rustls v0.14.1 (*)
│   ├── tokio v0.2.22 (*)
│   ├── tokio-rustls v0.13.1
│   │   ├── tokio v0.2.22 (*)
│   │   ├── tokio v0.2.22 (*)
│   │   ├── tokio v0.2.22 (*)
├── tokio v0.2.22 (*)
    ├── tokio v0.2.22 (*)
    ├── tokio-tungstenite v0.11.0
    │   ├── tokio v0.2.22 (*)

**Platform**
19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64

**Description**
I'm using warp for web-based application and all my routes are declared at the main function which grows to 100 lines of code.

When I tried to call the endpoints which are at bottom of the function this throws the error 
'tokio-runtime-worker' has overflowed its stack

**Comment by Darksonn** [comment, marked as answer]
Does it also overflow in release mode? Does it happen with Rust version 1.45? Can you post the function?

**Reply by 25sumitkr** [reply]
It works now by running the application on release mode.

Could you please help me to understand why it is not working on debug mode. Also, please confirm that if adding more endpoints this will not break the release mode as well.

Thanks for the quick reply. 🙏

**Reply by Darksonn** [reply]
Well it's because optimizations have reduced the amount of stack space necessary. Looking at your code, I can't help but think that there must be a better way. I would open an issue on the `warp` repository to ask for assistance.

**Comment by carllerche** [comment]
This is unlikely a tokio issue. The `tokio-runtime-worker` thread is where your application code runs.

If you believe it is a bug w/ Tokio, we will need more information to validate. Until then, I will convert this to a discussion.

**Reply by 25sumitkr** [reply]
I've pasted the code in the below link please find it
https://notepad.pw/143p2meo

I've taken the implementation reference from this github project
https://github.com/zupzup/rust-web-mongodb-example/blob/main/src/main.rs
